src/test/java/org/openmrs/contrib/qaframework/automation/LocationsSteps.java

```diff
@@ -0,0 +1,176 @@
+		manageLocationsPage.selectLocationTag();
+	}
+
+	@And("a user clicks on save location button")
+	public void clickOnSaveLocationButton() {
+		manageLocationsPage.saveLocation();
+	}
+
+	@Then("the system adds location into the locations table")
+	public void systemAddsLocation() {
+		assertTrue(textExists("Location Saved"));
+	}
+
+	@When("a user clicks on manage location tags link from configure metadata page")
+	public void goToManageLocationTagsPage() {
+		manageLocationTagsPage = metadataPage.goToManageLocationTagPage();
+	}
+
+	@Then("the system loads manage location tags page")
+	public void systemLoadsManageLocationTagsPage() {
+		assertTrue(textExists("Manage Location Tags"));
+	}
+
+	@And("a user clicks on add new location tag button")
+	public void goToLocationTagsForm() {
+		manageLocationTagsPage.goToAddNewLocationTagForm();
+	}
+
+	@Then("the system loads add new location tag form")
+	public void systemLoadsLocationTagsForm() {
+		assertTrue(textExists("Add New Location Tag"));
+	}
+
+	@And("a user enters location tag details into the form")
+	public void enterLocationTagDetails() {
+		manageLocationTagsPage.enterLocationTagName("Maternity Ward Wing");
+		manageLocationTagsPage.enterLocationTagDescription("For expectant mothers");
+	}
+
+	@And("a user clicks on save location tag button")
+	public void clickOnSaveLocationTagsButton() {
+		manageLocationTagsPage.saveLocationTag();
+	}
+
+	@Then("the system adds location tag into the location tags table")
+	public void systemAddsLocationTag() {
+		assertTrue(textExists("Location Tag Saved"));
+	}
+
+	@When("a user clicks on manage location attribute types link from configure metadata page")
+	public void goToManageLocationAttributeTypesPage() {
+		attributePage = metadataPage.goToManageLocationAttributeTypesPage();
+	}
+
+	@Then("the system loads manage location attribute types page")
+	public void systemLoadsManageLocationAttributeTypesPage() {
+		assertTrue(textExists("Manage Location Attribute Types"));
+	}
+
+	@And("a user clicks on add new location attribute type button")
+	public void goToManageLocationAttributeTypeForm() {
+		attributePage.goToAddNewLocationAttributeTypeForm();
+	}
+
+	@Then("the system loads add new location attribute type form")
+	public void systemLoadsManageLocationAttributeTypeForm() {
+		assertTrue(textExists("Add New Location Attribute Type"));
+	}
+
+	@And("a user enters location attribute type details into the form")
+	public void enterLocationAttributeTypeDetails() {
+		attributePage.enterLocationAttributeTypeName("Female Ward");
+		attributePage.enterLocationAttributeTypeDescription(DESCRIPTION);
+		attributePage.enterMinimumOccurs("1");
+		attributePage.enterMaximumOccurs("3");
+		attributePage.selectDatatype("Location Datatype");
+		attributePage.selectPreferredHandler(PREFERRED_HANDLER);
+	}
+
+	@And("a user clicks on save location attribute type button")
+	public void clickOnSaveLocationAttributeTypeButton() {
```

Comment:
Method naming would sound more cleaner as  clickOnSaveButton since by default it saves location in the behind scenes